skimage/io/sift.py

```diff
@@ -51,10 +59,13 @@ def _sift_read(f, mode='SIFT'):
         datatype = np.dtype([('column', float), ('row', float),
                              ('second_moment', (float, 3)),
                              ('sign', float), ('data', (float, feature_len))])
+
     data = np.fromfile(f, sep=' ')
     if data.size != nr_features * datatype.itemsize / np.dtype(float).itemsize:
         raise IOError("Invalid %s feature file." % mode)
 
+    f.close()
```

Comment:
I imagine it can be a potential bug. the docstring says that f can be a string. Therefore, it will throw an AttributeError exception.